auth/src/main/java/com/firebase/ui/auth/IdpResponse.java

```diff
@@ -195,7 +199,19 @@ public void writeToParcel(Parcel dest, int flags) {
         dest.writeParcelable(mUser, flags);
         dest.writeString(mToken);
         dest.writeString(mSecret);
-        dest.writeSerializable(mException);
+
+        try {
+            ObjectOutputStream oos = new ObjectOutputStream(new ByteArrayOutputStream());
+            oos.writeObject(mException);
+            oos.close();
```

Comment:
So does the exception come from `writeObject`?  If so should you close the stream in a `finally` block?

Reply:
Whoops, good catch! TBH, I straight copied that from the Android Source so tisk tisk to me for blindly copypastaing and 😕 to the Android team. 😆 Them not closing the stream probably explains why the Parcel was corrupted when I tried just catching the exception they threw.